### PR TITLE
feat: add support for checking specific applications

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/installation-proxy/index.js [hunk 1]
```diff
@@ -50,13 +52,54 @@ class InstallationProxyService {
...
+    }
+    if (opts.returnAttributes) {
+      request.ClientOptions.ReturnAttributes = opts.returnAttributes;
+    }
+
+    this.plistService.sendPlist(request);
+    const messages = await this.waitMessageCompletion();
+    for (const message of messages) {
+      if (message.LookupResult) {
+        return message.LookupResult;
```
Comment (imurchie) [inline lib/installation-proxy/index.js]: Will only one message ever have a `LookupResult`?

Comment (umutuzgur) [reply lib/installation-proxy/index.js]: Yes, that is correct. I could do `messages[0]. LookupResult` but I like the for loop check better

Comment (imurchie) [reply lib/installation-proxy/index.js]: Me too.

File: lib/installation-proxy/index.js [hunk 2]
```diff
@@ -50,13 +52,54 @@ class InstallationProxyService {
...
+   * @param {LookupApplicationOptions} opts the lookup options that wants to be passed
+   * @returns A map of the applications which the key is the bundleId
+   */
+  async lookupApplications (opts = {}) {
+    const request = {
+      Command: 'Lookup',
+      ClientOptions: {}
+    };
+    if (opts.bundleIds) {
+      request.ClientOptions.BundleIDs = _.isString(opts.bundleIds) ? [opts.bundleIds] : opts.bundleIds;
```
Comment (mykola-mokhnach) [inline lib/installation-proxy/index.js]: I'd rather check _.isArray

File: lib/installation-proxy/index.js [hunk 3]
```diff
@@ -50,13 +52,54 @@ class InstallationProxyService {
...
+    if (opts.applicationType) {
+      request.ClientOptions.ApplicationType = opts.applicationType;
+    }
+    if (opts.returnAttributes) {
+      request.ClientOptions.ReturnAttributes = opts.returnAttributes;
+    }
+
+    this.plistService.sendPlist(request);
+    const messages = await this.waitMessageCompletion();
+    for (const message of messages) {
```
Comment (mykola-mokhnach) [inline lib/installation-proxy/index.js]: Using `return filter.find || throw Error` might save some lines of code

**Pull request merged**